collector/therm_darwin.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !notherm
+
+package collector
+
+/*
+#cgo LDFLAGS: -framework IOKit -framework CoreFoundation
+#include <stdio.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+#include <IOKit/pwr_mgt/IOPMLib.h>
+#include <IOKit/pwr_mgt/IOPM.h>
+
+struct ref_with_ret {
+    CFDictionaryRef ref;
+    IOReturn ret;
+};
+
+struct ref_with_ret FetchThermal();
+
+struct ref_with_ret FetchThermal() {
+    CFDictionaryRef ref;
+    IOReturn ret;
+    ret = IOPMCopyCPUPowerStatus(&ref);
+    struct ref_with_ret result = {
+            ref,
+            ret,
+    };
+    return result;
+}
+*/
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"unsafe"
+)
+
+type thermCollector struct {
+	cpuSchedulerLimit typedDesc
+	cpuAvailableCPU   typedDesc
+	cpuSpeedLimit     typedDesc
+	logger            log.Logger
+}
+
+const therm = "therm"
```

Comment:
Is there non-cpu thermal information? Or is the CPU stuff just the first iteration?

Also, let's call this `"thermal"`

Reply:
OK, I will rename that. And here is only information for CPU currently.